src/app.js

```diff
@@ -1,16 +1,37 @@
 import React from "react";
 import ReactDOM from "react-dom";
+import {ErrorBoundary} from './components/ErrorBoundary';
+import {SearchPage} from './pages/SearchPage';
+import {DetailsPage} from './pages/DetailsPage';
 
+import 'normalize.css/normalize.css';
 import './styles.scss';
 
-const App = () => (
-  <div className="App">
-    <h1>Hello World</h1>
-    <p>
-      This is awesome that everything is
-      working so flawlessly
-    </p>
-  </div>
-);
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+
+    // For shocasing purposes
+    this.state = {
+      page: 'details'
+    };
+  }
+
+  onPageChange = (page) => {
+    this.setState({page});
+  }
+
+  render() {
+    return (
+      <div className="App">
+        <ErrorBoundary>
+          {this.state.page === 'search'? 
```

Comment:
This is fine.
Just want to add how usually I do this kind of conditionals and you can see if it improve or not readability.

```javascript
{this.state.page === 'search' && <SearchPage {...} />}
{this.state.page === 'details' && <DetailsPage {...} />}
```

Either way, is up to you :)

Really awesome that you're already thinking about routing.

If you want to just for curiosity get it to futher level, you can implement this using [Context API](https://reactjs.org/docs/context.html) from react